index.js

```diff
@@ -1,20 +1,302 @@
+    } else console.log("I don't understand");
+    answer = await ask("\n>_");
+  }
+
+  answer = await ask(beach1.description + "\n>_"); //description of next room, which prompts a user response. need to put in a loop or similar framework to be able to move back to previous rooms.
+
+  while (currentRoom === "beach1") {
+    if (answer === "move south") {
+      changeRoom("beach2");
+      break;
+    } else if (answer === "move north") {
+      changeRoom("boat");
+      break;
+    } else if (answer === "move east") {
+      console.log("The cliffs are too high to climb");
+    } else if (answer === "move west" && player.playerInv.includes("key")) {
+      changeRoom("lighthouse");
+      break;
+    } else if (answer === "move west" && !player.playerInv.includes("key")) {
+      console.log("The handle does not budge. Maybe there is a key nearby.");
+    } else console.log("I don't understand");
+    answer = await ask("\n>_");
+  }
+
+  answer = await ask(beach2.description + "\n>_");
+
+  while (currentRoom === "beach2") {
+    if (answer === "move east") {
+      changeRoom("cave1");
+      break;
+    } else if (answer === "move west" || answer === "move south") {
+      console.log("The ocean stretches too far for you to swim your way out");
+    } else if (answer === "move north") {
+      changeRoom("beach1");
+      break;
+    } else console.log("I don't understand");
+    answer = await ask("\n>_");
+  }
+
+  answer = await ask(cave1.description + "\n>_");
+
+  while (currentRoom === "cave1") {
+    if (answer === "move south") {
+      changeRoom("cave2");
+      break;
+    } else if (answer === "move north" || answer === "move east") {
+      console.log("You cannot leave the cave this way");
+    } else if (answer === "move west") {
+      changeRoom("beach2");
+      break;
+    } else console.log("I don't understand");
+    answer = await ask("\n>_");
+  }
+
+  answer = await ask(cave2.description + "\n>_");
+
+  while (currentRoom === "cave2") {
+    if (
+      answer === "move south" ||
+      answer === "move east" ||
+      answer === "move west"
+    ) {
+      console.log("You cannot leave the cave this way");
+    } else if (answer === "move north") {
+      changeRoom("cave1");
+      break;
+    } else if (answer === "inspect skeleton") {
+      console.log(
+        "The skeleton must have been here for years. It is completely decomposed. In its left hand is a folded letter. The small amount of light in the room reveals a key in the coat's front pocket."
+      );
+    } else if (answer === "take key") {
+      player.playerInv.push(lighthouseKey);
+      cave2.inventory.pop(lighthouseKey);
+      console.log(lighthouseKey.description);
+    } else if (answer === "take letter") {
+      player.playerInv.push(letter);
+      cave2.inventory.pop(letter);
+    } else if (answer === "read letter") {
+      player.playerInv.push(letter);
+      cave2.inventory.pop(letter);
+      console.log(letter.description);
```

Comment:
All of this functionality worked for grabbing items into my inventory, good use of .push and .pop